sugarloaf/src/font/mod.rs

```diff
@@ -906,25 +895,26 @@ fn find_font_path(
 }
 
 #[cfg(not(target_arch = "wasm32"))]
-fn load_from_font_source(path: &PathBuf) -> Option<Arc<Vec<u8>>> {
+fn load_from_font_source(path: &PathBuf) -> Option<SharedData> {
     use std::io::Read;
 
     let cache = get_font_data_cache();
-    let mut cache_guard = cache.lock();
 
-    // Check if already cached - avoids repeated disk I/O
-    if let Some(cached_data) = cache_guard.get(path) {
-        return Some(Arc::clone(cached_data));
+    // Check if already cached - DashMap handles concurrent access efficiently
+    if let Some(cached_data) = cache.get(path) {
+        return Some(cached_data.clone());
     }
 
     // Load from disk if not cached
     if let Ok(mut file) = std::fs::File::open(path) {
         let mut font_data = vec![];
         if file.read_to_end(&mut font_data).is_ok() {
-            // Store in cache for future use
-            let font_data_arc = Arc::new(font_data);
-            cache_guard.insert(path.clone(), Arc::clone(&font_data_arc));
-            return Some(font_data_arc);
+            let shared_data = SharedData::new(font_data);
+            // Use entry API to handle concurrent inserts properly
+            let entry = cache
+                .entry(path.clone())
+                .or_insert_with(|| shared_data.clone());
+            return Some(entry.clone());
```

Comment:
This makes sense to me. I was gonna suggest just inserting since that guarantees you're getting the most recent version of the data, but this allows you to potentially avoid a write to the map, which is nice. So I'm chill with this.

Reply:
Yea, just trying to avoid write as possible but frankly as we talked idk even if it will be like oh so much is happening 😆